app/models/user.rb

```diff
@@ -2,13 +2,16 @@ class User < ApplicationRecord
 
   EMAIL_REGEXP = /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i.freeze
 
-  after_destroy :ensure_an_admin_remains
   validates :name, presence: true, uniqueness: true
   has_secure_password
   validates :email, uniqueness: true, allow_nil: true
   validates :email, format: {
     with: EMAIL_REGEXP
   }
+  after_create :send_welcome_mail
+  around_update :ensure_not_admin
+  around_destroy :ensure_not_admin
```

Comment:
why around?

Reply:
because I want to check while updating or deleting.
we also use before and  in this, I also use throw :abort to stop the execution.

Reply:
done